esp/services/ws_store/espstorelib/daliKVStore.cpp

```diff
@@ -166,7 +166,7 @@ IPropertyTree * CDALIKVStore::getAllKeyProperties(const char * storename, const
 
     xpath.appendf("/%s/%s", ns, key);
 
-    Owned<IRemoteConnection> conn = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT_KVSTORE);
+    Owned<IRemoteConnection> conn = querySDS().connect(xpath.str(), myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_KVSTORE);
     if (!conn)
         throw MakeStringException(-1, "DALI Keystore fetchKeyProperties(): Unable to connect to DALI KeyValue store path '%s'", xpath.str()); //rodrigo, not sure if this is too much info
 
```

Comment:
not related, but :
```
    Owned<IPropertyTree> keytree = conn->getRoot();
    if (!keytree.get())
        throw MakeStringException(-1, "DALI KV Store fetchKeyProperties(): Unable to access key '%s'", key); //this store doesn't exist
```

isn't necessary. If a connection succeeds, it has a root by definition. Also not necessary to link the root, i.e. would be more efficient/conventional to do :
```
    return conn->getRoot();
```

as getPropTree(".") is just self, so same as root.